### ISSUE TITLE
Enhancement : Recalbox P2K ?

### ISSUE THREAD

**Issue by soaresden** [Voljega/ExoDOSConverter#26]
Since the new version of Recalbox called Recalbox 7, each game can be remapped with a small file called P2k (Pad2Keyboard)

Example : "DukeNukemII.pc.p2k.cfg" which is the Foldername + .p2k.cfg

It would be Great to generate this file with the conversion with a "default" scheme.
It would be Awesome to Give a personnalized layout for each game (too much work ?)
Since 80% use CTRL+ALT+SPACE ENTER and Escape, and arrow keys, I think the default one is a good choice

Below is the inside of the file:

```
# Recalbox's Pad-to-Keyboard configuration
#NAME OF THE GAME - Duke Nukem 2
#DOS
 
# dpad to arrow keys
0:up = up
0:down = down
0:right = right
0:left = left
0:j1up = up
0:j1down = down
0:j1right = right
0:j1left = left

# Start is enter
0:start = enter ;; Entree
# Select is escape
0:select = esc ;; Back
# X is space
0:b = space
# O is ctrl
0:a = leftctrl ;; Jump
# /\ is nothing
0:x = 
# [] is Alt
0:y = leftalt ;; Fire

#L1 is F1
0:l1 = F1 ;; Game Options
#L2 is H
0:l2 = H ;; Help

#R1 is F5
0:r1 = F5
#R2 is B
0:r2 = b
#L3 is N
0:l3 = n ;; No (N)
#R3 is Y
0:r3 = y ;; Yes (Y)

#hotkey to quit DosBox
0:hotkey = F9 ;; Hotkey with reset (CTRL+F9)

#Format to Respect : 
#n°PAD (0 as default):ACTION = KEYBOARD KEY
#https://gblobscdn.gitbook.com/assets%2F-LdKWTKrrUvJVmGP83hw%2F-M6aDDoaznN3j6JMGFqb%2F-M6aDPL6keVccv58ghs5%2FP2K-layout-v2.png?alt=media&token=e1f7524c-9ada-4083-a854-b66410d5a74e

#Joystick Actions Available: 
#up			Digital PAD
#down		Digital PAD
#left		        Digital PAD
#right		Digital PAD
#j1up		Joystick #1 
#j1down		Joystick #1
#j1left		Joystick #1 
#j1right	        Joystick #1
#a			Bouton A
#b			Bouton B
#x			Bouton X
#y			Bouton Y
#start		Bouton START
#select		Bouton SELECT
#l1			Bouton L1
#l2			Bouton L2
#r1			Bouton R1
#r2			Bouton R2
#l3			Bouton L3
#r3			Bouton R3
#hotkey		Bouton HOTKEY
#j2up		Joystick #2 (droite), direction haute
#j2down		Joystick #2 (droite), direction basse
#j2left		Joystick #2 (droite), direction gauche
#j2right	        Joystick #2 (droite), direction droite
```

**Comment by Voljega** [comment]
I can do that no problem, but the formatting of github messes up the content of the file, can you repost the file content on pastebin ?

**Comment by soaresden** [comment]
here the file

it's a textfile, so rename it p2k.cfg instead of txt 👍 
[P2K template.txt](https://github.com/Voljega/ExoDOSConverter/files/5364527/P2K.template.txt)

**Comment by Voljega** [comment]
thank you

so the DukeNukemII.pc.p2k.cfg file must be at the same level as the DukeNukemII.pc folder right? not inside it ?

for the personnalized layout there two competing options :
- from what I understood they should be on screenscrapper, and will be downloaded when you scrap
- there is an alternate project called keyb2joypad which wants to map all dos games command, it's mostly a Google Sheet at the time, so not mature at all, when the project advance it could be translated to recabox format

**Comment by soaresden** [comment]
Yes, the same level.

Yes, normally people like me upload their files to screenscrapper. But don't mess with it. When people scrap their games, they will take it too (if avalaible), and overwrite the default one you integer?

**Comment by Voljega** [comment]
yep i was saying that for custom ones

**Comment by Voljega** [comment]
@soaresden Compatibility has been added and is parted of the just released 0.6-beta

**Comment by Voljega** [comment]
@soaresden  If you can tell me if it works, that would be nice, thanks :)
